### PR TITLE
feat: Add reward functions for voting in oracle module

### PR DESCRIPTION
**Pull request by justyntemme**
## Description

Adds reward mechanism for oracle voting 

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by toteki — APPROVED** [review]

File: x/oracle/alias_functions.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	"github.com/umee-network/umee/x/oracle/types"
+)
+
+// GetOracleAccount returns oracle ModuleAccount
+func (k Keeper) GetOracleAccount(ctx sdk.Context) authtypes.ModuleAccountI {
```
Comment (alexanderbez) [inline x/oracle/alias_functions.go:11]: I like this auxiliary method @toteki -- I think we should apply it to the x/leverage module as well when needing to get the module account.

File: x/oracle/reward.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/umee-network/umee/x/oracle/types"
+)
+
+// RewardBallotWinners implements
+// at the end of every VotePeriod, we give out portion of seigniorage reward(reward-weight) to the
+// oracle voters that voted faithfully.
```
Comment (alexanderbez) [inline x/oracle/reward.go:12]:
```suggestion
// RewardBallotWinners is executed at the end of every voting period, where we
// give out a portion of seigniorage reward(reward-weight) to the oracle voters
// that voted correctly.
```

File: x/oracle/reward.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	rewardDenoms[0] = types.MicroUmeeDenom
+
+	i := 1
+	for denom := range voteTargets {
+		rewardDenoms[i] = denom
+		i++
+	}
+
+	// Sum weight of the claims
+	ballotPowerSum := int64(0)
```
Comment (alexanderbez) [inline x/oracle/reward.go:30]:
```suggestion
	var ballotPowerSum int64
```

File: x/oracle/reward.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+			continue
+		}
+
+		periodRewards = periodRewards.Add(sdk.NewDecCoinFromDec(
+			denom,
+			sdk.NewDecFromInt(rewardPool.Amount).Mul(distributionRatio),
+		))
+	}
+
+	// Dole out rewards
```
Comment (alexanderbez) [inline x/oracle/reward.go:57]:
```suggestion
	// distribute rewards
```

File: x/oracle/reward.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		if receiverVal != nil && !rewardCoins.IsZero() {
+			k.distrKeeper.AllocateTokensToValidator(ctx, receiverVal, sdk.NewDecCoinsFromCoins(rewardCoins...))
+			distributedReward = distributedReward.Add(rewardCoins...)
+		}
+	}
+
+	// Move distributed reward to distribution module
+	err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, k.distrName, distributedReward)
+	if err != nil {
+		panic(fmt.Sprintf("[oracle] Failed to send coins to distribution module %s", err.Error()))
```
Comment (alexanderbez) [inline x/oracle/reward.go:75]:
```suggestion
		panic(fmt.Sprintf("failed to send coins to distribution module %s", err.Error()))
```

**Review by alexanderbez — COMMENTED** [review]
Nice work @justyntemme

**Review by alexanderbez — APPROVED** [review]
🎉

**Pull request merged**